cmd/podman/containers/create.go

```diff
@@ -298,6 +298,10 @@ func CreateInit(c *cobra.Command, vals entities.ContainerCreateOptions, isInfra
 		vals.Entrypoint = &val
 	}
 
+	if !isInfra && c.Flag("domainname").Changed && !c.Flag("hostname").Changed {
+		return vals, fmt.Errorf("%w: cannot specify domainname without hostname", define.ErrInvalidArg)
+	}
+
```

Comment:
This does not match docker

Reply:
> I took a quick look at docker and this doe snot match it. The domain name is added in /etc/hosts and `setdomainname()`? Looks like they jusy use a sysctl for that: https://github.com/moby/moby/blob/master/daemon/oci_utils.go#L8
> 
> Also you are ignoring the compat API value and it must be added to podman container inspect.

that sysctl just takes care of the `setdomainname` aspect right? we still have to write to /etc/hosts as we do with other host vals?

Reply:
ahhhh we can't use the sysctl as rootless, you get: `Error: crun: open /proc/sys/kernel/domainname: Permission denied: OCI permission denied`

Reply:
Interesting setdomainname() seems to work:
```
unshare -r -u sh -c 'domainname test && domainname'
```

Reply:
@Luap99 see: https://github.com/moby/moby/pull/41840

we might need to make a crun change if we want to do it this way.

do you think we could just handle it manually rather than using the sysctl?

Reply:
sounds like something that would need to get added to the oci spec so that all oci runtimes have to do it

Manually calling setdomainname is possible but we would have to join the container uts namespace which is always difficult to do properly in go.

Reply:
https://github.com/containers/crun/blob/b66340c93cc8e8a5a8148ea840221b2cb663d896/tests/test_start.py#L186-L216 makes me think @giuseppe might have some reservations with this.. crun seems to think this should not be allowed

Reply:
Not at sure but I think this can part of OCI spec, spec can have a special field and let runtime process this after setting up UTS namespace as edge-case or a small hack in runtime  could be just create/setup UTS namespace after `kernel.domainname` as being discussed here ( https://github.com/containers/crun/pull/983 ) but second one sounds like hack.

Reply:
so we'd have to modify the OCI spec? @flouthoc